File: Python For Everyone, Horstmann/Chapter3-Decisions/P3.14.py
# Write a program that takes user input describing a playing card in the following
# shorthand notation:
#     A Ace
#     2 ... 10  Card values
#     J Jack
#     Q Queen
#     K King
#     D Diamonds
#     H Hearts
#     S Spades
#     C Clubs
# Your program should print the full description of the card. For example,
#     Enter the card notation: QS
#     Queen of Spades

from sys import exit

card = str(input("Enter the notation for the card: "))

valueCode = ""
suitCode = ""

if len(card) == 3:
    valueCode = card[:2]
    suitCode = card[2:3]

else:
    valueCode = card[:1]
    suitCode = card[1:2]

value = ""

if valueCode == "A":
    value = "Ace"

elif valueCode == "K":
    value = "King"

elif valueCode == "Q":
    value = "Queen"

elif valueCode == "J":
    value = "Jack"

elif int(valueCode) >= 2 and int(valueCode) <= 10:
    value = valueCode

else:
    exit("Unknown card. Exiting...")

suit = ""

if suitCode == "H":
    suit = "Hearts"

elif suitCode == "D":
    suit = "Diamonds"

elif suitCode == "C":
    suit = "Clubs"

elif suitCode == "S":
    suit ="Spades"

print("That card is the %s of %s" % (value, suit))

File: Big-Java-for-Everyone-Late-Objects-Horstmann/Chapter-3-Decisions/P3_14.java
/*
 * Write a program that takes user input describing a playing card in the following shorthand
 * notation: A Ace 2 ... 10 Card values J Jack Q Queen K King D Diamonds H Hearts S Spades C Clubs
 * Your program should print the full description of the card. For example, Enter the card notation:
 * QS Queen of Spades
 */

import java.util.Scanner;

public class P3_14 {
  public static void main(String[] args) {
    Scanner input = new Scanner(System.in);
    System.out.print("Your card: ");
    String card = input.next();
    input.close();
    String value = card.substring(0, card.length() - 1);
    String suit = card.substring(card.length() - 1);
    String card_name = "";

    if (value.compareTo("A") == 0) {
      card_name = "Ace";
    } else if (value.compareTo("J") == 0) {
      card_name = "Jack";
    } else if (value.compareTo("Q") == 0) {
      card_name = "Queen";
    } else if (value.compareTo("K") == 0) {
      card_name = "King";
    } else {
      card_name = value;
    }

    if (suit.compareTo("D") == 0) {
      card_name += " of Diamonds";
    } else if (suit.compareTo("H") == 0) {
      card_name += " of Hearts";
    } else if (suit.compareTo("S") == 0) {
      card_name += " of Spades";
    } else if (suit.compareTo("C") == 0) {
      card_name += " of Clubs";
    }
    System.out.println(card_name);
  }
}
